code/headers/display_adapter.hpp

```diff
@@ -61,13 +89,282 @@ namespace r2d2::display {
          * @param data
          */
         virtual void set_pixels(uint16_t x, uint16_t y, uint16_t width,
-                                uint16_t height, const uint16_t data);
+                                uint16_t height, const uint16_t data) {
+            // set all the pixels to data
+            for (std::size_t t_y = 0; t_y < width; t_y++) {
+                for (std::size_t t_x = 0; t_x < width; t_x++) {
+                    // set the pixel with datas
+                    set_pixel(x + t_x, y + t_y, data);
+                }
+            }
+        }
+
+        /**
+         * @brief Sets character in a single color
+         *
+         * @param x x-coordinate of the character (x=0 is the leftmost collumn)
+         * @param y y-coordinate of the character (y=0 is the highest row)
+         * @param character The un-extended (0-127) ascii value of the character
+         * @param pixel_color The color of the character
+         */
+        virtual void set_character(uint16_t x, uint16_t y, char character,
+                                   uint16_t pixel_color) {
+            // Loops through all rows of the character
+            const hwlib::image &character_image = display_font[character];
```

Comment:
I dont see why a color array would help the set_pixels function calls set_pixel just as many times so it wont be more efficient

Reply:
My reasoning is, is that a screen that implements `set_pixels` could handle all these pixels at the same time instead of going pixel by pixel. That would be more efficient for a unbuffered screen for example.